api/pkg/service/catalog/syncer.go

```diff
@@ -262,8 +262,10 @@ func (s *syncer) updateResources(
 
 		rerr.Combine(s.updateResourceCategory(txn, log, &dbRes, r.Categories))
 		s.updateResourceTags(txn, log, &dbRes, r.Tags)
-		s.updateResourceVersions(txn, log, catalog, dbRes.ID, r.Versions)
-
+		// platform ids on resource version level
+		verPlatformIds := map[uint]bool{}
+		s.updateResourceVersions(txn, log, catalog, dbRes.ID, r.Versions, &verPlatformIds)
```

Comment:
if we add platforms only for resource versions then I think we can fetch platforms for a resource from its versions.
i think in gorm queries is we can use https://gorm.io/docs/preload.html#Nested-Preloading and then get platforms for a resource

Reply:
but this also means we will have to loop on all resource versions everytime we want to fetch resource details. I am okay with adding for both but wdyt ? which would be better?

Reply:
I tried several approaches here. 
The first idea was to have 1 join table for resources, versions and platforms, however at the end the problem was to get unique platforms for resources, because by default I got the same values several times in case of simple Preload, and Distinct was not possible to add to the default Preload, and at the end I thought that it's too complicated modifications in too many places.
The second idea was similar to your suggestion with getting resource platforms from resource versions' platforms, and at the end I decided not to implement it, because some queries work on resource level only, others are designed to return result of one db.Select and we would really need to extend many requests and loop through all resource versions all the time. So I thought that's too much overhead here.
That's why I chose the third way - separate storage of platforms for resources and platforms for resource versions. As we need to go through resource versions only when there are changes on catalog side and write or delete is done only when it's necessary. + the new logic is isolated.

However here I'd rely on your experience and project roadmap understanding. What is your gut feeling? :)

Reply:
oh yeah in first case there would be redundancy 
>db.Select and we would really need to extend many requests and loop through all resource versions all the time. So I thought that's too much overhead here.

yep that's was my concern

>However here I'd rely on your experience and project roadmap understanding. What is your gut feeling? :)

I would like to keep the current that's seems better, we can improve it later if found a better way :)